index.html

```diff
@@ -917,40 +825,35 @@ <h2>
         checkbox.setAttribute("type", "checkbox");
         document.body.appendChild(checkbox);
 
-        const lock = new WakeLock("screen");
+        let controller;
 
-        lock.onactivechange = () =&gt; checkbox.checked = lock.active;
+        checkbox.onclick = () =&gt; {
+          try {
+            checkbox.disabled = true;
 
-        const controller = new AbortController();
-
-        try {
-          checkbox.disabled = true;
-          if (checkbox.checked) {
-            await lock.request({ signal: controller.signal });
-          } else {
-            controller.abort();
+            if (checkbox.checked) {
+              controller = new AbortController();
+              WakeLock.request("screen", { signal: controller.signal });
+            } else if (controller) {
+              controller.abort();
+            }
+          } catch {
+            checkbox.checked = false;
+          } finally {
+            checkbox.disabled = false;
```

Comment:
The checkbox disabled state should be removed from this example since it is now no longer possible to determine when a wake lock has gone from requested to acquired.

Reply:
Fixed